### PR TITLE
Detection_event class introduced

### PR DESCRIPTION
**Pull request by stefanodellosa-personal**
In order to better converge all detection episode information, such as images path, camera id, time, etc., a new dedicated class has been introduced.

### CODE REVIEW

File: wadas/domain/detection_event.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    def __init__(
+        self,
+        camera_id,
+        time_stamp,
+        original_image,
+        detection_img_path,
+        detected_animals,  # List of detected animals
+        classification=True,
+        classification_img_path=None,
+        classified_animals=None,  # List of classified animals
```
Comment (cdimauro) [inline wadas/domain/detection_event.py:20]: Redundant comments.

File: wadas/domain/ftp_camera.py [hunk 2]
```diff
@@ -11,14 +11,14 @@
...
-    def __init__(self, id, ftp_folder, ftp_user, enabled=True, actuators=None):
+    def __init__(self, id, ftp_folder, enabled=True, actuators=None):
         if actuators is None:
             actuators = []
         super().__init__(id, enabled)
         self.type = Camera.CameraTypes.FTP_CAMERA
         self.ftp_folder = ftp_folder
-        self.user = ftp_user
         self.actuators = actuators
+        "Note: Camera ID is also FTP user"
```
Comment (cdimauro) [inline wadas/domain/ftp_camera.py:21]: This should be a comment.

File: wadas/domain/operation_mode.py [hunk 3]
```diff
@@ -93,6 +95,32 @@ def _detect(self, cur_image_path):
...
 
+    def _format_classified_animals_string(self, classified_animals):
+        # Prepare a list of classified animals to print in UI
+        self.last_classified_animals_str = ""
+        for animal in classified_animals:
+            last = animal["classification"][0]
+            if not self.last_classified_animals_str:
+                self.last_classified_animals_str = self.last_classified_animals_str + last
+            else:
+                self.last_classified_animals_str = self.last_classified_animals_str + ", " + last
```
Comment (cdimauro) [inline wadas/domain/operation_mode.py:106]: self.last_classified_animals_str = ", ".join(animal["classification"][0] for animal in classified_animals)

File: wadas/domain/operation_mode.py [hunk 4]
```diff
@@ -93,6 +95,32 @@ def _detect(self, cur_image_path):
...
+
+    def _classify(self, cur_image_path, detection_results):
+        """Method to run the animal classification process
+        on a specific image starting from the detection output"""
+        # Classify if detection has identified animals
+        if len(detection_results["detections"].xyxy) > 0:
```
Comment (cdimauro) [inline wadas/domain/operation_mode.py:112]: if len(detection_results["detections"].xyxy):

File: wadas/domain/test_model_mode.py [hunk 5]
```diff
@@ -46,28 +51,24 @@ def run(self):
         # Classify if detection has identified animals
         if len(det_results["detections"].xyxy) > 0:
```
Comment (cdimauro) [inline wadas/domain/test_model_mode.py:52]: if len(det_results["detections"].xyxy):

File: wadas/domain/whatsapp_notifier.py [hunk 6]
```diff
@@ -87,7 +94,11 @@ def send_whatsapp_message(self, img_path):
                 if message_response.status_code == 200:
                     logger.info("WhatsApp notification sent!")
                 else:
-                    logger.error(message_response.status_code, message_response.text)
+                    err_message = (
+                        f"Failed to send WhatsApp notification: "
```
Comment (cdimauro) [inline wadas/domain/whatsapp_notifier.py:98]: Redundant f-string (nothing is interpolated inside the string).

File: wadas/ui/configure_ftp_cameras_dialog.py [hunk 7]
```diff
@@ -15,9 +15,10 @@
     QLineEdit,
     QRadioButton,
     QScrollArea,
-    QWidget, QMessageBox,
+    QWidget,
+    QCheckBox
```
Comment (cdimauro) [inline wadas/ui/configure_ftp_cameras_dialog.py:19]: Sort imports.

File: wadas/domain/whatsapp_notifier.py (hunk 6 above)
Comment (stefanodellosa-personal) [reply wadas/domain/whatsapp_notifier.py:98]: it's a multiline f string (due to line too long fix):

                    err_message = (
                        f"Failed to send WhatsApp notification: "
                        f"{message_response.status_code}, {message_response.text}"
                    )

Comment (cdimauro) [reply wadas/domain/whatsapp_notifier.py:98]: Yes, but actually only the second one needs to interpolate the string and a value: the first one is just a plain string, and it doesn't use the interpolation abilities of f-strings.

File: wadas/ui/mainwindow.py [hunk 8]
```diff
@@ -398,10 +398,13 @@ def check_notification_enablement(self):
...
+            message_box.setWindowTitle("No enabled notification protocol")
+            message_box.setText(message)
+            message_box.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+            answer = message_box.exec()
 
-            if answer == message_box.No:
+            if answer == QMessageBox.No:
                 return False
             else:
                 return True
```
Comment (cdimauro) [inline wadas/ui/mainwindow.py:410]: return answer != QMessageBox.No:

**Review by hexfati — APPROVED** [review]
LGTM

**Pull request merged**